docs/ops/infrastructure/Loop_5.md

```diff
@@ -0,0 +1,181 @@
+## Loop <a name="Loop"></a> {#openvino_docs_ops_infrastructure_Loop_5}
+
+**Versioned name**: *Loop-5*
+
+**Category**: Infrastructure
+
+**Short description**: *Loop* operation performs recurrent execution of the network, which is described in the `body`, iterating through the data. 
+The operation has similar semantic to the ONNX* Loop [operation](https://github.com/onnx/onnx/blob/master/docs/Changelog.md#Loop-13).
+
+**Detailed description**
+
+The body of the Loop can be executed 0 or more times depending on the values passed to the Loop operation inputs called "trip count", "execution condition" and input of the Loop body called "current iteration".
+
+These Loop operation inputs have the following meaning:
+1. Trip count is an integer scalar or 1D tensor with 1 element input specifying maximum number of iterations. To simulate infinite loop Constant `-1` can be provided as input.
+2. Loop execution condition input is a boolean scalar or 1D tensor with 1 element input specifying whether to run the first loop iteration or not. Note, that the body of the Loop must yield the condition value for the consecutive iterations.
+
+There are several combinations of these two inputs `(trip_count, execution condition)` which are described in the following code snippet:
+
+```
+  input (-1, true) // infinite loop
+      bool cond = true;
+      for (int i = 0; cond; ++i) 
+      {
+          cond = true; // sub-graph calculating condition must always return "true"!
+      }
+
+  input (-1, cond) // while loop
+      bool cond = ...;
+      for (int i = 0; cond; ++i) 
+      {
+          cond = ...;
+      }
+
+  input (-1, true) // do-while loop
+      bool cond = true;
+      for (int i = 0; cond; ++i) 
+      {
+          cond = ...;
+      }
+
+  input (trip_count, true) // for loop
+      int trip_count = ...;
+      bool cond = true;
+      for (int i = 0; i < trip_count; ++i) 
+      {
+          cond = true; // sub-graph calculating condition must always return "true"!
+      }
+
+  input (trip_count, cond) // for with condition
+      int trip_count = ...;
+      bool cond = ...;
+      for (int i = 0; i < trip_count && cond; ++i) 
+      {
+          cond = ...;
+      }
+```
+
+1. One of the body graph inputs called "current iteration" is an integer scalar or 1D integer tensor with 1 number specifying current iteration number. The iteration number starts from 0 and incremented by one for each iteration. This input is optional and may not exist if the iteration number value is not used in the body.
+2. One of the body graph outputs is called "condition" is a boolean scalar or 1D tensor with 1 element. This value is used to decide whenever to perform the next iteration or not.
```

Comment:
Based on https://github.com/openvinotoolkit/openvino/pull/2583#discussion_r503383461, should it be always "First body graph output..."?

Reply:
We have decided during internal discussions to make this output to be in arbitrary position. The reason is that in some cases the sub-graph calculating the condition may be removed from the body. And since plugins don't support optional outputs properly the easies way is to specify this type of output using the attribute.